### PR TITLE
Propagate CodeActionOptions through fix all

### PR DESCRIPTION
**Pull request by tmat**

### CODE REVIEW

**Review by JoeRobich — APPROVED** [review]

File: src/EditorFeatures/Core.Wpf/Suggestions/FixAll/FixMultipleOccurrencesService.cs [hunk 1]
```diff
@@ -22,25 +24,28 @@ namespace Microsoft.CodeAnalysis.Editor.Implementation.Suggestions
...
-    internal class FixMultipleOccurrencesService : IFixMultipleOccurrencesService
+    internal sealed class FixMultipleOccurrencesService : IFixMultipleOccurrencesService
     {
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public FixMultipleOccurrencesService(IAsynchronousOperationListenerProvider listenerProvider)
-            => listenerProvider.GetListener(FeatureAttribute.LightBulb);
+        public FixMultipleOccurrencesService(IGlobalOptionService globalOptions, IAsynchronousOperationListenerProvider listenerProvider)
+        {
+            listenerProvider.GetListener(FeatureAttribute.LightBulb);
```
Comment (sharwell) [inline src/EditorFeatures/Core.Wpf/Suggestions/FixAll/FixMultipleOccurrencesService.cs:33]: 😕 What's going on here?

File: src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/FixAllContext.cs [hunk 2]
```diff
@@ -85,17 +85,22 @@ public partial class FixAllContext
...
             FixAllScope scope,
-            string codeActionEquivalenceKey,
+            string? codeActionEquivalenceKey,
             IEnumerable<string> diagnosticIds,
             DiagnosticProvider fixAllDiagnosticProvider,
             CancellationToken cancellationToken)
-            : this(new FixAllState(null, document, codeFixProvider, scope, codeActionEquivalenceKey, diagnosticIds, fixAllDiagnosticProvider),
+            : this(new FixAllState(
+                    fixAllProvider: null,
+                    document ?? throw new ArgumentNullException(nameof(document)),
```
Comment (sharwell) [inline src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/FixAllContext.cs:94]: 💡 Use `!!` instead

Comment (tmat) [reply src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/FixAllContext.cs:94]: I tried but getting syntax formatter warnings :(.

File: src/EditorFeatures/Core.Wpf/Suggestions/FixAll/FixMultipleOccurrencesService.cs (hunk 1 above)
Comment (tmat) [reply src/EditorFeatures/Core.Wpf/Suggestions/FixAll/FixMultipleOccurrencesService.cs:33]: Changed the approach a bit and forgot to clean up.

Comment (tmat) [reply src/EditorFeatures/Core.Wpf/Suggestions/FixAll/FixMultipleOccurrencesService.cs:33]: image

File: src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/FixAllContext.cs (hunk 2 above)
Comment (tmat) [reply src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/FixAllContext.cs:94]: image

Comment (tmat) [reply src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/FixAllContext.cs:94]: Also interesting - there can't be  whitespace between the identifier and `!!` :-o
image

Comment (tmat) [reply src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/FixAllContext.cs:94]: Filed https://github.com/dotnet/roslyn/issues/59580

**Pull request merged**